homeassistant/components/homekit/util.py

```diff
@@ -572,3 +572,11 @@ def state_needs_accessory_mode(state: State) -> bool:
         and state.attributes.get(ATTR_SUPPORTED_FEATURES, 0)
         & RemoteEntityFeature.ACTIVITY
     )
+
+
+def state_changed_event_is_same_state(event: Event) -> bool:
+    """Check if a state changed event is the same state."""
+    event_data = event.data
+    old_state: State | None = event_data.get("old_state")
+    new_state: State | None = event_data.get("new_state")
+    return bool(new_state and old_state and new_state.state == old_state.state)
```

Comment:
Not perse better, but `last_changed` only updates when the state updates. `last_updated` always updates.

```suggestion
    return bool(new_state and old_state and new_state.last_changed != old_state.last_updated)
```

Reply:
I guess the attributes could change as well so probably best to keep it as state == state ?

Reply:
Tried that change and the test fails so left it as-is and added a test for the attributes changing as well